lib/json_api_resource/resource.rb

```diff
@@ -54,7 +54,13 @@ def method_missing(method, *args, &block)
       if match = method.to_s.match(/^(.*)=$/)
         self.client.send(match[1], args.first)
       elsif self.client.respond_to?(method.to_sym)
-        self.client.send(method)
+        is_method = self.client.methods.include?(method.to_sym)
+        argument_count = is_method ? self.client.method(method.to_sym).arity : 0
+        if argument_count > 0 || argument_count == -1
```

Comment:
can we invert this to

```ruby
if client.method(method.to_sym).arity == 0
  client.send(method)
else
  client.send(method, args.first)
end
```
and for brevity I think we can remove the `self`, not sure what I was running into when I built this.